src/System.Memory/src/System/Buffers/Text/Base64Decoder.cs

```diff
@@ -29,26 +34,48 @@ public static partial class Base64
         /// </summary> 
         public static OperationStatus DecodeFromUtf8(ReadOnlySpan<byte> utf8, Span<byte> bytes, out int bytesConsumed, out int bytesWritten, bool isFinalBlock = true)
         {
+            // PERF: use uint to avoid the sign-extensions
+            uint sourceIndex = 0;
+            uint destIndex = 0;
+
+            if (utf8.IsEmpty)
+                goto DoneExit;
+
             ref byte srcBytes = ref MemoryMarshal.GetReference(utf8);
             ref byte destBytes = ref MemoryMarshal.GetReference(bytes);
 
             int srcLength = utf8.Length & ~0x3;  // only decode input up to the closest multiple of 4.
             int destLength = bytes.Length;
+            int maxSrcLength = srcLength;
+            int decodedLength = GetMaxDecodedFromUtf8Length(srcLength);
 
-            int sourceIndex = 0;
-            int destIndex = 0;
+            // max. 2 padding chars
+            if (destLength + 2 < decodedLength)
```

Comment:
destLength + 2 could integer overflow and result in a negative value being compared. This doesn't appear to be handled properly later in the method.

Reply:
I fixed this -- thanks.

Even without the fix this should not be a problem, as https://github.com/dotnet/corefx/blob/8f7b490ca874ee2a9f11f0163412f7c95811298b/src/System.Memory/src/System/Buffers/Text/Base64Decoder.cs#L57, but so it is "more correct" and safer anyways. 👍